src/librustc_trans/back/link.rs

```diff
@@ -288,7 +288,7 @@ pub fn mangle<PI: Iterator<Item=PathElem>>(path: PI,
     // when using unix's linker. Perhaps one day when we just use a linker from LLVM
     // we won't need to do this name mangling. The problem with name mangling is
     // that it seriously limits the available characters. For example we can't
-    // have things like &T or ~[T] in symbol names when one would theoretically
+    // have things like &T or Vec<T> in symbol names when one would theoretically
```

Comment:
Not sure this change makes sense. I think it's referring to the `~` more than it's referring to the fact that it's a vector.